### PR TITLE
add new helper to display completion on learner dashboard

### PR DESCRIPTION
**Pull request by melvinsoft**
## Change description

> Description here

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Comment by melvinsoft** [comment]
@OmarIthawi This is the PR we were talking about, I think we can afford the tech debt without testing, since it's going to be removed in Maple. 
Please let me know.

**Comment by OmarIthawi** [comment]
@melvinsoft here's a proposed way to test this pull request: 

  - Make a test case that creates a course with minimal content.
  - Create a learner (User) that's enrolled in that course
  - Run this function and expect it to return no completion

The benefits of this approach:

 - We cover all the lines of the code in this function mitigating against basic syntax and other errors 
 - Make sure this function at least return sane results for someone enrolling in a course
 - The function works for the releases: Juniper, Koa ... Maple
 - It's relatively cheap to 

The downsides:

 - We need to spend a bit more effort on manual testing
 - We don't really know how the function works after the system changes, whether it's an upgrade or any other change

But, given what we have at hand we probably don't want to spend effort on properly testing this function with completion since we have to focus on Tahoe 2.0.

**Comment by melvinsoft** [comment]
@OmarIthawi It's hard to give up 😞 . I could not make those test work. I did an extensive review on the code in `edx/master` and is doesn't have a test either, here is the original PR: https://github.com/openedx/edx-platform/pull/26928

Then I did a search in the code: https://github.com/openedx/edx-platform/search?q=get_course_blocks_completion_summary

There isn't any test either.

We don't like un tested code, but my reasoning here is, we are porting a function from edX which is going to be used in a single place for now, after we upgrade to Maple, we'll remove it and use the upstream one, which is un tested too.

File: openedx/core/djangoapps/appsembler/html_certificates/helpers.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+def get_course_blocks_completion_summary(course_key, user):
+    """
+    Returns an object with the number of complete units, incomplete units, and units that contain gated content
+    for the given course. The complete and incomplete counts only reflect units that are able to be completed by
+    the given user. If a unit contains gated content, it is not counted towards the incomplete count.
+    The object contains fields: complete_count, incomplete_count, locked_count
+    """
+    if not user.id:
+        return []
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/html_certificates/helpers.py:19]:
```suggestion
        return {}
```

File: openedx/core/djangoapps/appsembler/html_certificates/helpers.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Appsembler Helpers to improve course info in learner dashboard.
+We should remove this after Maple, since all the info is in the new course_home_api.
+"""
+
+from xmodule.modulestore.django import modulestore
+from common.djangoapps.student.helpers import cert_info
+from lms.djangoapps.course_blocks.api import get_course_blocks
+
+
+def get_course_blocks_completion_summary(course_key, user):
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/html_certificates/helpers.py:11]:
```suggestion
@beeline.traced('html_certificates.get_course_blocks_completion_summary')
def get_course_blocks_completion_summary(course_key, user):
```

File: openedx/core/djangoapps/appsembler/html_certificates/helpers.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    """
+    Returns an object with the number of complete units, incomplete units, and units that contain gated content
+    for the given course. The complete and incomplete counts only reflect units that are able to be completed by
+    the given user. If a unit contains gated content, it is not counted towards the incomplete count.
+    The object contains fields: complete_count, incomplete_count, locked_count
+    """
+    if not user.id:
+        return []
+    store = modulestore()
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/html_certificates/helpers.py:20]:
```suggestion
    beeline.add_context_field('username', user.username)
    beeline.add_context_field('course_key', course_key)
    store = modulestore()
```

**Review by OmarIthawi — APPROVED** [review]
Thanks @melvinsoft. Yeah, I see no tests either which is a bumper. The original PR branch is called `ciduarte/hack` which makes more sense now.

Please fix the `{}` empty return and add honeycomb tracing since I think this would be a somewhat slow function when learners have many courses.

**Review by OmarIthawi — APPROVED** [review]

**Comment by OmarIthawi** [comment]
@melvinsoft the tests are broken for some reason. Please try to fix them and let me now if there's anything I can help with.

Also please squash the commits into a single one so it's cleaner in the history.

**Pull request merged**